### PR TITLE
Validate SpvOpVectorShuffle

### PR DESCRIPTION
**Pull request by jozefkucia**

### CODE REVIEW

File: source/validate_id.cpp [hunk 1]
```diff
@@ -1701,12 +1701,41 @@ bool idUsage::isValid<OpVectorInsertDynamic>(
...
+  // Component operands.
+  auto componentCount = inst->words.size() - 5;
+  auto resultVectorDimension = resultType->words()[3];
+  if (componentCount != resultVectorDimension) {
+    DIAG(inst->words.size() - 1)
+        << instr_name() << " component literals count does not match "
+                           "Result Type <id> '"
+        << resultType->id() << "'s vector component count.";
+    return false;
+  }
```
Comment (dneto0) [inline source/validate_id.cpp]: This is good as far as it goes.
But there are more rules to check:
- each literal is either 0xFFFFFFFF,
   or as an unsigned integer at most  the sum of the lengths of the input vectors, minus 1

Comment (jozefkucia) [reply source/validate_id.cpp]: Thanks for the review. I've added checks for the remaining rules.

**Comment by dneto0** [comment]
Waiting for the bots to pass

**Comment by dneto0** [comment]
One bot failed spuriously.  I kicked it off again.

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as 90862fe4b1c6

**Pull request closed without merging**